packages/app/components/token-picker/TokenPicker.tsx

```diff
@@ -1,69 +1,105 @@
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  Pepe
-                </div>
-                <div className="p-2 font-semibold uppercase border text-em-med w-fit rounded-2xl">
-                  usdc
-                </div>
-              </div>
-            </div>
-            <div className="relative">
-              <div className="-mx-4 overflow-y-auto border-t h-72 border-surface-50">
-                {[
-                  "1Inch",
-                  "Aave",
-                  "DAI",
-                  "USDC",
-                  "Matic",
-                  "Uni",
-                  "pepe",
-                  "Swapr",
-                  "ARB",
-                ].map((token) => (
-                  <div
-                    className="flex justify-between w-full px-4 py-2 border-b cursor-pointer border-surface-50 hover:bg-surface-50"
-                    key={token}
-                    onClick={closeAction}
-                  >
-                    <div className="flex items-center space-x-3">
-                      <div className="w-8 h-8 rounded-full bg-primary-100" />
-                      <BodyText>{token}</BodyText>
-                    </div>
-                    <BodyText>0</BodyText>
-                  </div>
-                ))}
-              </div>
-            </div>
-          </div>
-        </ModalContent>
-      </Modal>
-    </div>
+    <Modal closeAction={closeAction} isOpen={isOpen}>
+      <ModalHeaderTitle closeAction={closeAction} title="Select a token" />
+      <ModalContent>
+        <SearchBar />
+        <CommonTokens />
+        <TokenList closeAction={closeAction} tokenList={tokenList} />
+      </ModalContent>
+    </Modal>
   );
 };
+
+const SearchBar = () => (
+  <div className="flex items-center bg-surface-50 border border-surface-75 rounded-xl py-2 px-3 text-em-low mx-6">
+    <Icon className="text-em-med" name="search" size={18} />
+    <input
+      className="outline-none font-semibold flex-grow ml-2 bg-surface-50"
+      placeholder="Search token name or paste address"
+      type="text"
+    />
+  </div>
+);
+
+const CommonTokens = () => (
+  <div className="mt-5 px-6">
+    <p className="text-xs font-semibold text-em-low">Common tokens</p>
+    <div className="flex flex-wrap mt-2 gap-2">
+      {TOKEN_PICKER_COMMON_TOKENS.map((token: Token) => (
+        <RadioButton
+          checked={false}
+          id={`${token.address}${token.symbol}`}
+          key={`${token.address}${token.symbol}`}
+          name={token.name}
+          onChange={() => {}}
+          value={token.name}
+        >
+          <TokenIcon token={token} />
+          <p className="font-semibold ml-1.5">{token.symbol}</p>
+        </RadioButton>
+      ))}
+    </div>
+  </div>
+);
+
+const TokenListRow = ({ closeAction, token }: TokenListRowProps) => (
+  <div
+    className="flex justify-between w-full px-6 py-2 cursor-pointer hover:bg-surface-50"
```

Comment:
I think we can remove the padding x from elements.

```suggestion
    className="flex justify-between w-full py-2 cursor-pointer hover:bg-surface-50"
```

Reply:
It would result in something like this
![image](https://github.com/SwaprHQ/stackly-ui/assets/21271189/63831609-db10-46c0-9d8d-ba0c3668e106)

I like it better TBH, I had it like this before but tried to replicate the designs. Then we had the chat about not following 100% the designs, if we can deviate a bit, I really like this display more